packages/orbit-components/src/TextLink/__tests__/index.test.js

```diff
@@ -1,73 +1,68 @@
 
-describe("TextLink", () => {
-  const title = "My text link";
-  const href = "https://kiwi.com";
-  const onClick = jest.fn();
-  const type = "primary";
-  const tabIndex = "-1";
-  const dataTest = "test";
-  const rel = "nofollow";
+const title = "My text link";
+const dataTest = "test";
 
-  const component = shallow(
-    <TextLink
-      onClick={onClick}
-      href={href}
-      type={type}
-      rel={rel}
-      external
-      icon={<ChevronRight />}
-      tabIndex={tabIndex}
-      dataTest={dataTest}
-    >
-      {title}
-    </TextLink>,
-  );
-
-  const componentWithoutHref = shallow(<TextLink>{title}</TextLink>);
-
-  it("should contain a children", () => {
-    expect(component.children().exists()).toBe(true);
-  });
-  it("should contain an href", () => {
-    expect(component.render().prop("href")).toBe(href);
-  });
-  it("should have data-test", () => {
-    expect(component.render().prop("data-test")).toBe(dataTest);
-  });
-  it("should have noopener in attribute", () => {
-    expect(component.render().prop("rel").split(" ").includes("noopener")).toBe(true);
-  });
-  it("should have noreferrer in attribute", () => {
-    expect(component.render().prop("rel").split(" ").includes("noreferrer")).toBe(true);
-  });
-  it("should have rel values in the rel attribute", () => {
-    expect(component.render().prop("rel").split(" ").includes(rel)).toBe(true);
-  });
-  it("should have tabindex", () => {
-    expect(component.render().prop("tabindex")).toBe(tabIndex);
-  });
-  it("should contain an external href", () => {
-    expect(component.render().prop("target")).toBe("_blank");
-  });
-  it("should contain SVG", () => {
-    expect(component.find("ChevronRight").exists()).toBe(true);
+describe("#TextLink", () => {
+  it("should be focusable and have button role", () => {
+    render(<TextLink>{title}</TextLink>);
+    userEvent.tab();
+    expect(screen.getByText(title)).toHaveFocus();
+    expect(screen.getByRole("button")).toBeInTheDocument();
   });
+
   it("should execute onClick method", () => {
-    component.simulate("click");
+    const onClick = jest.fn();
+    render(<TextLink onClick={onClick}>{title}</TextLink>);
+    userEvent.click(screen.getByText(title));
     expect(onClick).toHaveBeenCalled();
   });
-  it("should not have tabindex and role", () => {
-    expect(component.render().prop("tabindex")).toBe("-1");
-    expect(component.render().prop("role")).toBe(undefined);
+
+  it("should render with props", () => {
+    const dataTestIcon = "testIcon";
+    const tabIndex = "-1";
+    const href = "https://kiwi.com";
+    render(
+      <TextLink
+        href={href}
+        icon={<ChevronRight dataTest={dataTestIcon} />}
+        tabIndex={tabIndex}
+        dataTest={dataTest}
+      >
+        {title}
+      </TextLink>,
+    );
+    expect(screen.getByText(title)).toBeInTheDocument();
+    expect(screen.getByText(title).closest("a")).toHaveAttribute("href", href);
```

Comment:
Seems like this should be enough:
```suggestion
    expect(screen.getByRole("link")).toHaveAttribute("href", href);
```